src/core/execute.c

```diff
@@ -1849,6 +1849,43 @@ char** exec_context_get_restrict_filesystems(const ExecContext *c) {
         return l ? TAKE_PTR(l) : strv_new(NULL);
 }
 
+static int extension_name_is_vpicked(const char *path) {
+        _cleanup_free_ char *dir = NULL;
+        int r;
+
+        assert(path);
+
+        r = path_extract_filename(path, &dir);
+        if (r == -EADDRNOTAVAIL)
+                return 0;
+        if (r < 0)
+                return r;
+
+        return endswith(dir, ".v") ? 1 : 0;
+}
+
+int exec_context_has_vpicked_extensions(const ExecContext *context) {
+        int r;
+
+        assert(context);
+
+        if (context->n_extension_images == 0 && strv_isempty(context->extension_directories))
+                return 0;
```

Comment:
tese two lines are redundant, and are not going to bring any improvement. (they don#t hurt either, admittedly, but I'd always remove such redundancies if it helps neither performance nor readability